### PR TITLE
fix #22606: gammasimp (y|func(y)) + gamma(x))

### LINKED ISSUE #22606: Inconsistent behaviour of simplifying the gamma function
The following example shows an inconsistency when `gamma` is simplified:

```
from sympy import gamma, symbols, Function
a, s, t = symbols('a s t')
f = Function('f')
x = a+gamma(s)
y = f(t)+gamma(s)
print('%s\n%s'%(x.simplify(), y.simplify()))
```

Output:
```
a + gamma(s)
f(t) + gamma(s + 1)/s
```

While it is correct, I do not see why `gamma` should be simplified like this when added to an undefined function.

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

#22606 

#### Brief description of what is fixed or changed

One~~Two~~ issue resolved:
1) divergence of behavior for `x + gamma(y)` vs `f(x) + gamma(y)`
~~2) returning of `(x + gamma(x + 1))/x` instead of `1 + gamma(x)` for `gammasimp(1 + gamma(x))`; the inverse expression is also handled~~

#### Release Notes

* simplify
    * the presence of UndefinedFunction no longer affects how expressions with gamma functions are simplified

### CODE REVIEW

File: sympy/simplify/gammasimp.py [hunk 1]
```diff
@@ -139,6 +150,29 @@ def gamma_factor(x):
                 return gamma_factor(x.base)
             return False
 
+        # distribution pre-step
+        if level == -1:
+            n, d = fraction(expr, exact=True)
+            def distr(d, n):
+                # divide and Add arg in n by symbolic d and return
```
Comment (smichr) [inline sympy/simplify/gammasimp.py:157]:
```suggestion
                # divide an Add arg in n by symbolic d and return
```

File: sympy/integrals/tests/test_integrals.py [hunk 2]
```diff
@@ -1558,7 +1558,7 @@ def test_issue_15285():
 def test_issue_15432():
     assert integrate(x**n * exp(-x) * log(x), (x, 0, oo)).gammasimp() == Piecewise(
         (gamma(n + 1)*polygamma(0, n) + gamma(n + 1)/n, re(n) + 1 > 0),
-        (Integral(x**n*exp(-x)*log(x), (x, 0, oo)), True))
+        (Integral(x**n*exp(-x)*log(x), (x, 0, oo)), True)),integrate(x**n * exp(-x) * log(x), (x, 0, oo)).gammasimp()
```
Comment (smichr) [inline sympy/integrals/tests/test_integrals.py:1561]:
```suggestion
        (Integral(x**n*exp(-x)*log(x), (x, 0, oo)), True))
```

**Comment by MishraAditya12126** [comment]
Sir i would love to work on this issue under the guidance of yours , if no one else is assigned do assign it to me @smichr

**Comment by smichr** [comment]
> Sir i would love to work on this issue

The work is this PR. If you want to review it or have a improved proposal, that would be fine.

**Comment by hanspi42** [comment]
@smichr: My PR that made me submit this issue has been merged; this means you will need to change 

https://github.com/sympy/sympy/blob/a3e0f9c810cb78d8ca68b128757551783564f218/sympy/integrals/tests/test_transforms.py#L41

back to

https://github.com/sympy/sympy/blob/65fd07d897ad1249bab6ff19de3d701a01fc65bc/sympy/integrals/tests/test_transforms.py#L41

**Pull request merged**